erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -37,6 +52,32 @@ def before_submit(self):
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)), title=_("Quantity Mismatch"))
 
+	def before_cancel(self):
+		#update picked_qty in SO Item on cancel of PL
+		for location in self.get('locations'):
+			if location.sales_order_item:
+				self.update_so(location.sales_order_item,0,location.item_code)
```

Comment:
This does _nothing_ on cancellation 😄 Picked qty remains as is since it's already there in already_picked. Here we need to reverse impact by adding negative qty to already picked qty.

Reply:
please add a test for this.

Reply:
reference.

Created SO, created PL, cancelled PL. Picked qty doesn't go back to zero. 

<img width="1142" alt="Screenshot 2022-03-17 at 6 48 04 PM" src="https://user-images.githubusercontent.com/9079960/158816851-4ff6f316-1183-44a0-bf0e-f1d4d6220bd7.png">

<img width="1142" alt="Screenshot 2022-03-17 at 6 48 12 PM" src="https://user-images.githubusercontent.com/9079960/158816888-c07e4059-942c-43da-8a3d-c311b33a12b3.png">